pkgs/development/go-modules/generic/default.nix

```diff
@@ -198,7 +198,7 @@ go.stdenv.mkDerivation (
      ln -s "${dep.src}" "$d/src/${dep.goPackagePath}"
   ''
   ) goPath) + ''
-    export GOPATH="$d:$GOPATH"
+    export GOPATH=${lib.concatStringsSep ":" ( ["$d"] ++ ["$GOPATH"] ++ ["$PWD"] ++ extraSrcPaths)}
```

Comment:
I don't understand how adding `$PWD` to the `GOPATH` would work. Is it because the project itself contains that `src/..` structure? If that's the case it seems to be project specific, most go projects assume to be located inside of $GOPATH or use the vendor/ folder